### PR TITLE
Moves Asmdiff to arcade

### PR DESCRIPTION
**Pull request by Anipik**
Fixes https://github.com/dotnet/runtime/issues/35025

### CODE REVIEW

File: src/Microsoft.DotNet.Asmdiff/Columns/DiffAssemblyCsvColumn.cs [hunk 1]
```diff
@@ -0,0 +1,66 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using Microsoft.Cci.Extensions;
+using Microsoft.Cci.Mappings;
+
+namespace Microsoft.DotNet.Asmdiff
```
Comment (ericstj) [inline src/Microsoft.DotNet.Asmdiff/Columns/DiffAssemblyCsvColumn.cs:9]: Can we change the namespace to match the subdirectory?  All of these types look to be specific to CSV so let's put them in their own namespace.  Not sure if "Columns" is best, consider "Csv" if more appropriate.

File: src/Microsoft.DotNet.Asmdiff/DiffComment.cs [hunk 2]
```diff
@@ -0,0 +1,21 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Microsoft.DotNet.Asmdiff
+{
+    public sealed class DiffComment
```
Comment (ericstj) [inline src/Microsoft.DotNet.Asmdiff/DiffComment.cs:7]: I wonder if this is useful in AsmDiff.

File: src/Microsoft.DotNet.Asmdiff/Columns/DiffAssemblyCsvColumn.cs (hunk 1 above)
Comment (ericstj) [reply src/Microsoft.DotNet.Asmdiff/Columns/DiffAssemblyCsvColumn.cs:9]: I wonder if we even care about this CSV output format.

File: src/Microsoft.DotNet.Asmdiff/DiffComment.cs (hunk 2 above)
Comment (Anipik) [reply src/Microsoft.DotNet.Asmdiff/DiffComment.cs:7]: no currently its just being as null,  the type is being used csharpwriter

**Comment by tmat** [comment]
Would you mind adding Readme.md to the directory that describes the tool a bit?

File: src/Microsoft.DotNet.Asmdiff/Columns/DiffAssemblyCsvColumn.cs (hunk 1 above)
Comment (Anipik) [reply src/Microsoft.DotNet.Asmdiff/Columns/DiffAssemblyCsvColumn.cs:9]: we can just add a markdownexporter and remove the csv stuff, but i suggest doing that in a separate change

**Comment by ViktorHofer** [comment]
Where are the sources coming from? Assuming it's from a git scm, can we please port the commits over as well?

File: src/Microsoft.DotNet.Asmdiff/Columns/DiffAssemblyCsvColumn.cs (hunk 1 above)
Comment (ericstj) [reply src/Microsoft.DotNet.Asmdiff/Columns/DiffAssemblyCsvColumn.cs]: At least give this a different namespace now, please.

File: src/Microsoft.DotNet.AsmDiff/Microsoft.DotNet.AsmDiff.csproj [hunk 3]
```diff
@@ -0,0 +1,16 @@
+<!-- Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE file in the project root for more information. -->
+<Project Sdk="Microsoft.NET.Sdk">
+  
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>netcoreapp3.1</TargetFramework>
```
Comment (ericstj) [inline src/Microsoft.DotNet.AsmDiff/Microsoft.DotNet.AsmDiff.csproj:6]: Should we cross-target this?  What are we doing for other tools like GenAPI and APICompat?

File: src/Microsoft.DotNet.Asmdiff/Columns/DiffAssemblyCsvColumn.cs (hunk 1 above)
Comment (Anipik) [reply src/Microsoft.DotNet.Asmdiff/Columns/DiffAssemblyCsvColumn.cs]: yes i have already done that.

**Comment by Anipik** [comment]
> Where are the sources coming from? Assuming it's from a git scm, can we please port the commits over as well?


Replied offline

**Comment by ericstj** [comment]
How does this get packaged?

**Comment by ViktorHofer** [comment]
> Replied offline

As noted offline, I believe we should bring the commit history over _if it's valuable_. I'm sure we can work out eventual security concerns.

File: src/Microsoft.DotNet.AsmDiff/Microsoft.DotNet.AsmDiff.csproj [hunk 4]
```diff
@@ -3,15 +3,17 @@
   
   <PropertyGroup>
     <OutputType>Exe</OutputType>
-    <TargetFrameworks>netcoreapp3.1;net472</TargetFrameworks>
+    <TargetFrameworks>netcoreapp3.1</TargetFrameworks>
+    <IsPackable>true</IsPackable>
+    <PackAsTool>true</PackAsTool>
+    <ToolCommandName>dotnet-asmdiff</ToolCommandName>
```
Comment (ericstj) [inline src/Microsoft.DotNet.AsmDiff/Microsoft.DotNet.AsmDiff.csproj:9]: Is it OK that this differs from the assembly name?

Comment (Anipik) [reply src/Microsoft.DotNet.AsmDiff/Microsoft.DotNet.AsmDiff.csproj:9]: we do a similar thing here https://github.com/dotnet/arcade/blob/master/src/Microsoft.DotNet.SwaggerGenerator/Microsoft.DotNet.SwaggerGenerator.CmdLine/Microsoft.DotNet.SwaggerGenerator.CmdLine.csproj#L9

so i kept it for consistency

Comment (alexperovich) [reply src/Microsoft.DotNet.AsmDiff/Microsoft.DotNet.AsmDiff.csproj:9]: ToolCommandName is just the name that `dotnet tool install` gives the shim that is added to the path when you install the tool.

**Comment by Anipik** [comment]
@ericstj @ViktorHofer Can we go ahead and merge this one ?

**Comment by ViktorHofer** [comment]
Do we have tests for the tool somewhere?

**Comment by Anipik** [comment]
> Do we have tests for the tool somewhere?

No we dont have any  tests for the tool

File: src/Microsoft.DotNet.AsmDiff/Microsoft.DotNet.AsmDiff.csproj [hunk 5]
```diff
@@ -0,0 +1,19 @@
+<!-- Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE file in the project root for more information. -->
+<Project Sdk="Microsoft.NET.Sdk">
+  
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFrameworks>netcoreapp3.1</TargetFrameworks>
```
Comment (ViktorHofer) [inline src/Microsoft.DotNet.AsmDiff/Microsoft.DotNet.AsmDiff.csproj:6]: nit: `<TargetFramework>netcoreapp3.1</TargetFramework>`

**Pull request merged**